akka-actor-typed/src/main/scala/akka/actor/typed/internal/receptionist/ServiceKeySerializer.scala

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.actor.typed.internal.receptionist
+
+import java.io.NotSerializableException
+import java.nio.charset.StandardCharsets
+
+import akka.actor.typed.receptionist.ServiceKey
+import akka.annotation.InternalApi
+import akka.serialization.{ BaseSerializer, SerializerWithStringManifest }
+
+@InternalApi
+class ServiceKeySerializer(val system: akka.actor.ExtendedActorSystem) extends SerializerWithStringManifest with BaseSerializer {
+
+  private val ServiceKeyManifest = "a"
+
+  def manifest(o: AnyRef): String = o match {
+    case _: ServiceKey[_] ⇒ ServiceKeyManifest
+    case _ ⇒
+      throw new IllegalArgumentException(s"Can't serialize object of type ${o.getClass} in [${getClass.getName}]")
+  }
+
+  def toBinary(o: AnyRef): Array[Byte] = o match {
+    case serviceKey: ServiceKey[_] ⇒ serviceKey.id.getBytes(StandardCharsets.UTF_8)
+    case _ ⇒
+      throw new IllegalArgumentException(s"Cannot serialize object of type [${o.getClass.getName}]")
+  }
+
+  def fromBinary(bytes: Array[Byte], manifest: String): ServiceKey[Any] = manifest match {
+    case ServiceKeyManifest ⇒ ServiceKey[Any](new String(bytes, StandardCharsets.UTF_8))
```

Comment:
`final case class DefaultServiceKey[T](id: String, typeName: String)`

i.e. it has the class name of the "Any" included in the `equals`
it is grabbed via a ClassTag

I think we said that we would like to include the `typeName: String` in the serialized format (as a String) and then this could be `new DefaultServiceKey(id, typeName) here.

You can use the typeName as the manifest.
You only have to care about `DefaultServiceKey` in the serializer. Make the serializer binding to that class.

Reply:
Added a new commit. 

Sorry if I didn't get it right, but please let me know if anything in my new commit still doesn't look good?